[Fix] npm install 시 styled-components@6.1.11와 react@18.2.0 사이의 의존성 충돌로 인해 발생한 오류 해결

## 📌 Issue
<!-- 이슈에 대한 설명 -->
- npm install 시 styled-components@6.1.11와 react@18.2.0 사이의 의존성 충돌로 인해 오류 발생
```
❯ npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: styled-components@6.1.11
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"18.2.0" from the root project
npm ERR!   peer react@"*" from @react-navigation/bottom-tabs@6.5.20
npm ERR!   node_modules/@react-navigation/bottom-tabs
npm ERR!     @react-navigation/bottom-tabs@"^6.5.20" from the root project
npm ERR!   12 more (@react-navigation/core, @react-navigation/elements, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react-dom@">= 16.8.0" from styled-components@6.1.11
npm ERR! node_modules/styled-components
npm ERR!   styled-components@"^6.1.11" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: react@18.3.1
npm ERR! node_modules/react
npm ERR!   peer react@"^18.3.1" from react-dom@18.3.1
npm ERR!   node_modules/react-dom
npm ERR!     peer react-dom@">= 16.8.0" from styled-components@6.1.11
npm ERR!     node_modules/styled-components
npm ERR!       styled-components@"^6.1.11" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

## 📜 To-do
<!-- 해야할 세부 Task -->
- [x] 각 version 최신으로 upgrade

## ETC
<!-- 그 외 기타 내용들 -->

--- a/.gitignore
+++ b/.gitignore
@@ -33,3 +33,6 @@ yarn-error.*
 
 # typescript
 *.tsbuildinfo
+
+# lock
+yarn.lock

--- a/package.json
+++ b/package.json
@@ -14,16 +14,16 @@
     "@react-navigation/native-stack": "^6.9.26",
     "expo": "~51.0.14",
     "expo-status-bar": "~1.12.1",
-    "react": "18.2.0",
+    "react": "18.3.1",
     "react-native": "0.74.2",
-    "react-native-safe-area-context": "4.10.1",
-    "react-native-screens": "3.31.1",
+    "react-native-safe-area-context": "4.10.5",
+    "react-native-screens": "3.32.0",
     "styled-components": "^6.1.11"
   },
   "devDependencies": {
-    "@babel/core": "^7.20.0",
-    "@types/react": "~18.2.45",
-    "typescript": "^5.1.3"
+    "@babel/core": "^7.24.7",
+    "@types/react": "~18.3.3",
+    "typescript": "^5.5.2"
   },
   "private": true
 }

